static/js/github-create-branch.js

```diff
@@ -1,47 +1,108 @@
+
+$(document).ready(() => {
+  $("#ghServers").auiSelect2();
+  $("#ghRepo").auiSelect2();
+  $("#ghParentBranch").auiSelect2({ data: [] });
+
+  $("#ghRepo").on("change", () => {
+    loadBranches();
+  });
+
+  $("#createBranchForm").on("aui-valid-submit", (event) => {
+    event.preventDefault();
+    createBranchPost();
+  });
+
+  $('#cancelBtn').click(function (event) {
+    event.preventDefault();
+    window.close();
+  });
+
+  // change event doesn't trigger for first time, so triggering it manually
+  $("#ghRepo").trigger("change");
+});
+
+const loadBranches = () => {
+  clearBranches();
+  toggleSubmitDisabled(true);
+  hideErrorMessage();
+  const repo = getRepoDetails();
+  $.ajax({
+    type: "GET",
+    url: `/github/create-branch/owners/${repo.owner}/repos/${repo.name}/branches`,
+    success: (data) => {
+      $("#ghParentBranch").auiSelect2({
+        data: () => {
+          data.repository.refs.edges.forEach((item) => {
+            item.id = item.node.name;
+          });
+          return {
+            text: (item) => item.node.name,
+            results: data.repository.refs.edges
+          }
+        },
+        formatSelection: (item) => { return item.node.name; },
```

Comment:
/nit
Can change this to a one liner
```suggestion
        formatSelection: (item) => item.node.name,
```